runtime/interpreter/interpreter.go

```diff
@@ -3646,6 +3621,92 @@ func (interpreter *Interpreter) IsSubType(subType DynamicType, superType sema.Ty
 	return false
 }
 
+func (interpreter *Interpreter) IsSubType(subType StaticType, superType StaticType) bool {
+	if superType == PrimitiveStaticTypeAny {
+		return true
+	}
+
+	// This is an optimization: If the static types are equal, then no need to check further.
+	// i.e: Saves the conversion time.
+	if subType.Equal(superType) {
+		return true
+	}
+
+	semaType, err := interpreter.ConvertStaticToSemaType(superType)
+	if err != nil {
+		return false
+	}
+
+	return interpreter.IsSubTypeOfSemaType(subType, semaType)
+}
+
+func (interpreter *Interpreter) IsSubTypeOfSemaType(subType StaticType, superType sema.Type) bool {
+	if superType == sema.AnyType {
+		return true
+	}
+
+	switch staticType := subType.(type) {
+	case PrimitiveStaticType:
+		if subType == PrimitiveStaticTypeString && superType == sema.CharacterType {
+			return true
+		}
+
+	case OptionalStaticType:
+		if typedSuperType, ok := superType.(*sema.OptionalType); ok {
+			return interpreter.IsSubTypeOfSemaType(staticType.Type, typedSuperType.Type)
+		}
+
+		switch superType {
+		case sema.AnyStructType, sema.AnyResourceType:
+			return interpreter.IsSubTypeOfSemaType(staticType.Type, superType)
+		}
```

Comment:
This is needed because: reference types at runtime behave differently from static-check. Hence optional types also need to use the runtime-behavior recursively.